holoviews/plotting/bokeh/callbacks.py

```diff
@@ -330,6 +350,75 @@ def _get_stream_handle_ids(self, handles):
         return stream_handle_ids
 
 
+    def on_change(self, attr, old, new):
+        """
+        Process change events adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._event_queue.append((attr, old, new))
+        if self.trigger not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.trigger, 50)
+
+
+    def on_event(self, event):
+        """
+        Process bokeh UIEvents adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._event_queue.append((event))
+        if self.trigger not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.trigger, 50)
+
+
+    def trigger(self):
+        """
+        Trigger callback change event and triggering corresponding streams.
+        """
+        if not self._event_queue:
+            return
+        if self.events:
+            event = self._event_queue[-1]
+        self._event_queue = []
+
+        values = {}
+        for attr, path in self.attributes.items():
+            attr_path = path.split('.')
+            if self.events:
+                obj = event
+                model_obj = self.plot_handles.get(self.models[0])
+            else:
+                if attr_path[0] == 'cb_obj':
+                    attr_path = self.models[:1]+attr_path[1:]
+                obj = self.plot_handles.get(attr_path[0])
+                model_obj = obj
+            attr_val = obj
+            if not obj:
+                raise Exception('Bokeh plot attribute %s could not be found' % path)
+            for p in attr_path[1:]:
+                if p == 'attributes':
+                    continue
+                if isinstance(attr_val, dict):
+                    attr_val = attr_val.get(p)
+                else:
+                    attr_val = getattr(attr_val, p, None)
+            values[attr] = {'id': model_obj.ref['id'],
+                            'value': attr_val}
```

Comment:
Given the suggestion of splitting trigger, maybe ``get_event_message(attributes)`` and ``get_change_message(attributes)``.